### ISSUE TITLE
Update action to use node 16

### ISSUE THREAD

**Issue by der-eismann** [aws-actions/configure-aws-credentials#489]
edit from the maintainers: This is now implemented! Please see [this comment](https://github.com/aws-actions/configure-aws-credentials/issues/489#issuecomment-1278145876) on how to update your action configuration. The original issue continues below.

---

Hello, thanks for the convenient action! Unfortunately it is a bit outdated, because it still uses the node12 runtime which isn't supported on self-hosted runners Apple Silicon / arm64 architecture.
Would you kindly update the action so it uses the [node16 runtime](https://github.blog/changelog/2022-05-20-actions-can-now-run-in-a-node-js-16-runtime/) instead?

**Comment by peterwoodworth** [comment]
Thanks for the request @der-eismann,

We'll start thinking about our next major version release so that we can support this 🙂

**Comment by lvkins** [comment]
+1

Github started giving deprecation annotations to my workflows containing this action:

> Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: aws-actions/configure-aws-credentials, aws-actions/configure-aws-credentials

**Comment by der-eismann** [comment]
There is already https://github.com/aws-actions/configure-aws-credentials/pull/413 open, it shouldn't be that hard.

**Comment by brt-eric-mccrea** [comment]
+1

**Comment by johnbeech** [comment]
Started seeing this warning as well; sure it's all over the place:

![image](https://user-images.githubusercontent.com/6341696/195063340-21f09c88-152d-4060-aae7-5cc281874b2e.png)

**Comment by just-Bri** [comment]
Would love to see [this PR](https://github.com/aws-actions/configure-aws-credentials/pull/520) get merged

**Comment by peterwoodworth** [comment]
Thanks everyone for letting us know about the need to upgrade the Node runtime for this action.

We cannot merge these PRs as they are unfortunately, because to make this change we need to do so on a new major version. The reason for this is that changing the Node version we could be breaking cases where the runners used don't have node 16 installed.

Since this issue has come up we have a strong need to release the next major version. We are working hard to deliver this, but there are a number of things that have to be done behind the scenes first. I will try to provide an ETA here when possible, thanks all for your patience 🙂

**Comment by sayhiben** [comment]
@peterwoodworth We're seeing this warning somewhere in the hundreds of times each day right now, and it creates a substantial amount of noise for our engineers. Please consider ways to release a Node 16 update ahead of other planned changes for the next release; perhaps an early major version with only this change is a possibility.

![image](https://user-images.githubusercontent.com/678890/195461179-b54aec35-3a4c-4edb-8634-15fffd4d0f83.png)

For what it's worth, my opinion is that GitHub could have more carefully thought through the way they provide deprecation warnings and done so in a way that wasn't so disruptive to normal development.

**Comment by mungojam** [comment]
> Thanks everyone for letting us know about the need to upgrade the Node runtime for this action.
> 
> We cannot merge these PRs as they are unfortunately, because to make this change we need to do so on a new major version. The reason for this is that changing the Node version we could be breaking cases where the runners used don't have node 16 installed.
> 
> Since this issue has come up we have a strong need to release the next major version. We are working hard to deliver this, but there are a number of things that have to be done behind the scenes first. I will try to provide an ETA here when possible, thanks all for your patience 🙂

#521 is a similar one that will cause warnings and will cause a breakage when they disable it

**Comment by pagameba** [comment]
While the planned change to node 16 is to take effect Summer 2023 according to the [deprecation notice](https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/), the deprecation warnings are disruptive by generating a lot of noise.  Piling this and a lot of new stuff into the next major version does not sound like quick process.  I think the suggestion of addressing this (and the ::set-output issue) in a new major version is excellent, assuming it is practical to separate out upcoming major version changes.

**Comment by TomBeckett** [comment]
@kellertk Do you know what the state is for other popular tasks like: [ECS-Render-Task-Definition](https://github.com/aws-actions/amazon-ecs-render-task-definition/releases)? 

Will there be similar branches available?

**Comment by kellertk** [comment]
> @kellertk Do you know what the state is for other popular tasks like: [ECS-Render-Task-Definition](https://github.com/aws-actions/amazon-ecs-render-task-definition/releases)?
> 
> Will there be similar branches available?

I'm not sure about that action, because a different AWS team is responsible for it. But I'll poke around internally and see if we can't get a better answer there.

**Comment by singhprd** [comment]
- Temporary bumping of aws-actions/amazon-ecr-login: https://github.com/fac/freeagent/pull/39968

**Comment by jensenbox** [comment]
Am I correct in thinking that once v2 is released that it will run Node 16?

I am totally OK with the error messages and being super ultra lazy, I rather just update once and only after the next release is ready.

**Comment by peterwoodworth** [comment]
v2 will run at least Node 16 😃

**Comment by leonk-sportsbet** [comment]
Is nodejs v16 coming for aws-actions/configure-aws-credentials@v1 ?

**Comment by peterwoodworth** [comment]
It's already here @leonk-sportsbet, but we have that on a separate branch `v1-node16`, so you will want to configure your action to use `aws-actions/configure-aws-credentials@v1-node16`. See [this comment](https://github.com/aws-actions/configure-aws-credentials/issues/489#issuecomment-1278145876) for more information on why and our future plans

**Comment by sterlingwes** [comment]
Thoughts on locking this issue for future comments? I think it's pretty clear what the plan is here and locking it will maybe force folks to read what's already been written.

I'm mostly interested when v2 lands and I imagine this issue being closed will signal that. Thank you contributors for being so responsive on this issue!

**Comment by kellertk** [comment]
Hi everyone, I'll be locking this thread for now (see the comment from @sterlingwes above for why). If you have any issues with the @v1-node16 branch please open a new issue. We'll close this issue when v2 is released.
